src/js/InfoBox.jsx

```diff
@@ -0,0 +1,132 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import './InfoBox.css';
+
+class InfoBox extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      currentPrice: null,
+      monthChangeD: null,
+      monthChangeP: null,
+      updatedAt: null
+    };
+  }
+  componentDidMount() {
+    this.getData();
+  }
+  shouldComponentUpdate(nextProps, nextState) {
+    return (
+      this.props.coin !== nextProps.coin ||
+      this.state.currentPrice !== nextState.currentPrice
+    );
+  }
+  componentWillUpdate() {
+    this.getData();
+  }
+  componentWillUnmount() {
+    clearInterval(this.refresh);
+  }
+
+  getData() {
+    const { data, coin, duration } = this.props;
+    const url = {
+      BTC: 'https://api.coindesk.com/v1/bpi/currentprice.json',
+      ETH: 'https://min-api.cryptocompare.com/data/price?fsym=ETH&tsyms=USD',
+      LTC: 'https://min-api.cryptocompare.com/data/price?fsym=LTC&tsyms=USD'
+    };
+
+    fetch(url[this.props.coin])
+      .then(r => r.json())
+      .then(bitcoinData => {
+        const price =
+          this.props.coin === 'BTC'
+            ? bitcoinData.bpi.USD.rate_float
+            : bitcoinData.USD;
+
+        const change = price - data[coin][duration][0].y;
+        const changeP =
+          (price - data[coin][duration][0].y) / data[coin][duration][0].y * 100;
+
+        const updatedAt =
+          this.props.coin === 'BTC'
+            ? bitcoinData.time.updated
+            : moment.utc().format('MMM D, YYYY hh:mm:ss');
+        this.setState({
+          currentPrice: price,
+          monthChangeD: change.toLocaleString('us-EN', {
+            style: 'currency',
+            currency: 'USD'
+          }),
+          monthChangeP: `${changeP.toFixed(2)} %`,
+          updatedAt
+        });
+      })
+      .catch(e => {
+        console.log(e);
```

Comment:
Are you intending to do some actual error handling?

Reply:
nope :+1: